frontend/libs/studio-components/src/components/StudioCombobox/StudioCombobox.test.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import { StudioCombobox, type StudioComboboxProps } from './index';
+import userEvent from '@testing-library/user-event';
+
+const options = {
+  ole: 'Ole',
+  dole: 'Dole',
+  doffen: 'Doffen',
+  invalid: 'Kinkliane Koff',
+};
+const noResults = 'No results';
+
+describe('StudioCombobox', () => {
+  const verifyOnValueChange = async (
+    onValueChange: jest.Mock,
+    expectedCalls: number,
+    expectedValue: string[],
+  ) => {
+    await waitFor(() => expect(onValueChange).toHaveBeenCalledTimes(expectedCalls));
+    expect(onValueChange).toHaveBeenCalledWith(expectedValue);
+  };
+
+  it('should render the component', () => {
+    renderTestCombobox();
+    const studioCombobox = screen.getByRole('combobox');
+    expect(studioCombobox).toBeInTheDocument();
+  });
+
+  it('should render the empty state when no results are found', async () => {
+    const user = userEvent.setup();
+    renderTestCombobox();
+    const studioCombobox = screen.getByRole('combobox');
+    await user.type(studioCombobox, options.invalid);
+    const emptyState = screen.getByText(noResults);
+    expect(emptyState).toBeInTheDocument();
+  });
+
+  it('should be possible to select an option', async () => {
+    const user = userEvent.setup();
+    const onValueChange = jest.fn();
+    renderTestCombobox({ onValueChange });
+    const studioCombobox = screen.getByRole('combobox');
+    expect(studioCombobox).toHaveValue('');
+
+    await user.click(studioCombobox);
+    await user.click(screen.getByRole('option', { name: options.ole }));
+    await verifyOnValueChange(onValueChange, 1, [options.ole]);
+    expect(studioCombobox).toHaveValue(options.ole);
+
+    const comboboxIsClosed = screen.queryByRole('option', { name: options.ole });
```

Comment:
When reading the name `comboboxIsClosed`, I think the value is a boolean. The query is the combobox option, but the result can be null. What about just naming it comboboxOption? 😊 

```suggestion
    const comboboxOption = screen.queryByRole('option', { name: options.ole });
```

Reply:
That is true, hm.. I had earlier a comment above explaining the reason for this check. But I thought I could remove it after changing the name to `comboboxIsClosed`.
The reason for this check is because of the bug I fixed in this PR where the combobox didn't close when selecting an option. 

I could go for the `comboboxOption`  that you suggest - but should I add a comment above it again or do you think it is not necessary?

Reply:
What about creating a separate test-case for that? `it('should close combobox when a option is selected')`. Then the test-case it self explains the reason. 😊

Reply:
Yeah, but it felt natural for me to keep it there between the open/close/open. 😄 But I agree with you, then the test-case explains it :) Will do that 👍